### ISSUE TITLE
Issue with password protected ssh files

### ISSUE THREAD

**Issue by EthanHindmarsh** [cargo-generate/cargo-generate#535]
Hello, I set up my `git` to use a password-protected ssh key w/ a `ssh-add` agent.
When trying to run cargo generate I get SSH errors.
```
[ethan ~/prgming/wasm-example]$ cargo generate --git https://github.com/rustwasm/wasm-pack-template

⚠   Unable to load config file: /home/ethan/.local/share/cargo/cargo-generate.toml
Error: ⛔   Git Error: authentication required but no callback set; class=Ssh (23); code=Auth (-16)

[ethan ~/prgming/wasm-example]$ cargo generate --git git@github.com:rustwasm/wasm-pack-template
⚠   Unable to load config file: /home/ethan/.local/share/cargo/cargo-generate.toml
💡   Using private key: `$HOME/.ssh/id_rsa` for git-ssh checkout
Error: ⛔   Git Error: Failed to authenticate SSH session: Unable to extract public key from private key file: Wrong passphrase or invalid/unrecognized private key file format; class=Ssh (23)
```

Here is my `~/.gitconfig`
<details>
<summary>gitconfig</summary>

```
[ethan ~/prgming/wasm-example]$ cat ~/.gitconfig

[url "ssh://git@github.com/"]
        insteadOf = https://github.com/
[user]
        name = EthanHindmarsh
        email = ethan@ethan.lol
        signingkey = 9D74F79E7386BF75501DEA8FE8325E9DA8304F95
[push]
        gpgSign = false
[gpg]
        program = gpg2
[receive]
        advertisePushOptions = true
[commit]
        gpgsign = true
[init]
        defaultBranch = master
```

</details>

I believe the 
```
[url "ssh://git@github.com/"]
        insteadOf = https://github.com/
```
is causing issues with the http url, and the password on my ssh key is causing issues with the ssh url

Is there any way for me to ignore my `.gitconfig` or use my `ssh-add` agent?

Thanks 👍

**Comment by sassman** [comment]
Hey @EthanHindmarsh,

Thanks for the detailed report.
So one thing is and remains a problem, that is protected ssh keys that are loaded via `ssh-add`, see #384 

The other issue with the `insteadOf` gitconfig is already adressed in #520 and will be released quite soon with `v0.13.0`
You can already beta test this on the main branch if you like to. 

Early feedback would indeed be very much appreciated.

**Comment by EthanHindmarsh** [comment]
> Hey @EthanHindmarsh,
> 
> Thanks for the detailed report. So one thing is and remains a problem, that is protected ssh keys that are loaded via `ssh-add`, see #384
> 
> The other issue with the `insteadOf` gitconfig is already adressed in #520 and will be released quite soon with `v0.13.0` You can already beta test this on the main branch if you like to.
> 
> Early feedback would indeed be very much appreciated.

Oh nice😁
I tried searching for either of these myself and didn't see any open issues at a glance, will close this now👍
